src/System.Net.WebSockets/src/System/Net/WebSockets/WebSocketException.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Net.WebSockets
+{
+    public sealed class WebSocketException : Win32Exception
```

Comment:
Interesting.  In the public contract, WebSocket derives from System.Exception and we "promote" the additional methods of Win32Exception to the class.  See below.  Perhaps we need to change this implementation for open source now to match that.

Here is the contract definition for the public WebSocketExceptionClass:

`
 public sealed partial class WebSocketException : System.Exception {
   public WebSocketException(int nativeError) { }
   public WebSocketException(int nativeError, System.Exception innerException) { }
   public WebSocketException(int nativeError, string message) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, System.Exception innerException) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, int nativeError) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, int nativeError, System.Exception innerException) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, int nativeError, string message) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, int nativeError, string message, System.Exception innerException) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, string message) { }
   public WebSocketException(System.Net.WebSockets.WebSocketError error, string message, System.Exception innerException) { }
   public WebSocketException(string message) { }
   public WebSocketException(string message, System.Exception innerException) { }
   public System.Net.WebSockets.WebSocketError WebSocketErrorCode { get { return default(System.Net.WebSockets.WebSocketError); } }
 }
`